pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperKitCreationService.java

```diff
@@ -0,0 +1,276 @@
+                        kitRequestSettings.getCollaboratorParticipantLengthOverwrite());
+
+        //In case it's needed, subkits handling should be added here
+
+        addJuniperKitRequest(conn, kitTypeName, kitRequestSettings, ddpInstance, kitType.getKitTypeId(), collaboratorParticipantId,
+                errorMessage, easyPostUtil, kit, externalOrderNumber, juniperKitRequestId, null, userId, transactionResults);
+
+        // Not needed now, uncomment later when needed for external shippers
+        //        orderKits.add(kit);
+
+        return transactionResults;
+    }
+
+    private SimpleResult addJuniperKitRequest(Connection conn, String kitTypeName, KitRequestSettings kitRequestSettings,
+                                              DDPInstance ddpInstance,
+                                              int kitTypeId, String collaboratorParticipantId, String errorMessage,
+                                              EasyPostUtil easyPostUtil,
+                                              JuniperKitRequest kit, String externalOrderNumber, String juniperKitRequestId,
+                                              String ddpLabel, String userId, SimpleResult transactionResults) {
+        String collaboratorSampleId = null;
+        String bspCollaboratorSampleType = kitTypeName;
+        String addressId = null;
+        try {
+            Address address = easyPostUtil.getAddress(kit.getEasypostAddressId());
+            if (address != null) {
+                addressId = address.getId();
+            }
+        } catch (EasyPostException e) {
+            throw new RuntimeException("EasyPost addressId could not be received ", e);
+        }
+
+        if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+            try {
+                // collaborator sample id is the id of the physical sample that GP will work with.
+                collaboratorSampleId =
+                        KitRequestShipping.generateBspSampleID(conn, collaboratorParticipantId, bspCollaboratorSampleType, kitTypeId);
+                KitRequestShipping.writeRequest(ddpInstance.getDdpInstanceId(), juniperKitRequestId, kitTypeId,
+                        kit.getParticipantId().trim(),
+                        collaboratorParticipantId, collaboratorSampleId, userId, addressId, errorMessage, externalOrderNumber, false,
+                        null, ddpInstance, bspCollaboratorSampleType, ddpLabel);
+                //             Not needed now
+                //            kit.setExternalOrderNumber(externalOrderNumber);
+            } catch (Exception e) {
+                transactionResults.resultException = e;
+            }
+        } else {
+            if (kitRequestSettings.getCollaboratorSampleTypeOverwrite() != null) {
+                bspCollaboratorSampleType = kitRequestSettings.getCollaboratorSampleTypeOverwrite();
+            }
+            if (StringUtils.isNotBlank(collaboratorParticipantId)) {
```

Comment:
Can you comment on what this case is? What is a `collaboratorParticipantId` and what does this code accomplish?